src/main/java/cinema/project/dao/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,76 @@
+package cinema.project.dao;
+
+import cinema.project.exception.DataProcessingException;
+import cinema.project.lib.Dao;
+import cinema.project.model.ShoppingCart;
+import cinema.project.model.User;
+import cinema.project.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+            return shoppingCart;
+        } catch (Exception ex) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add shopping cart: " + shoppingCart
+                    + " to the DB", ex);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> query =
+                    session.createQuery("SELECT s FROM ShoppingCart s"
+                            + " LEFT JOIN FETCH s.tickets t"
+                            + " LEFT JOIN FETCH t.movieSession m"
+                            + " LEFT JOIN FETCH m.movie"
+                            + " LEFT JOIN FETCH m.cinemaHall"
```

Comment:
do you need all of these joins?

Reply:
Yes, otherwise I get Lazy Initialization Exception